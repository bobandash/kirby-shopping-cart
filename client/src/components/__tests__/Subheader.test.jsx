import { describe, it, expect, beforeEach, vi } from "vitest";
import { fireEvent, render, screen } from "@testing-library/react";
import SubHeader from "../Subheader";
import { BrowserRouter } from "react-router-dom";
import userEvent from "@testing-library/user-event";

// setups props
describe("Subheader", () => {
  describe("DOM", () => {
    it("matches snapshot", () => {
      const routerWrapper = ({ children }) => <BrowserRouter>{children}</BrowserRouter>;
      const { container } = render(<SubHeader />, { wrapper: routerWrapper });
      expect(container).toMatchSnapshot();
    });
  });

  describe("Subheader Links", () => {
    beforeEach(() => {
      const routerWrapper = ({ children }) => <BrowserRouter>{children}</BrowserRouter>;
      render(<SubHeader />, { wrapper: routerWrapper });
    });

    it("Shop All works", async () => {
      const link = screen.getByRole("link", { name: "Shop All" });
      expect(link).toBeInTheDocument();
      const user = userEvent.setup();
      await user.click(link);
      expect(window.location.pathname).toBe("/category/all");
    });

    it("Games works", async () => {
      const link = screen.getByRole("link", { name: "Games" });
      expect(link).toBeInTheDocument();
      const user = userEvent.setup();
      await user.click(link);
      expect(window.location.pathname).toBe("/category/games");
    });

    it("Plushies works", async () => {
      const link = screen.getByRole("link", { name: "Plushies" });
      expect(link).toBeInTheDocument();
      const user = userEvent.setup();
      await user.click(link);
      expect(window.location.pathname).toBe("/category/plushies");
    });

    it("Keychains works", async () => {
      const link = screen.getByRole("link", { name: "Keychains" });
      expect(link).toBeInTheDocument();
      const user = userEvent.setup();
      await user.click(link);
      expect(window.location.pathname).toBe("/category/keychains");
    });
  });
});
