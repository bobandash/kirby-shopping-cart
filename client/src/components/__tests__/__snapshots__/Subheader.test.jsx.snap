// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > DOM > matches snapshot 1`] = `
<div>
  <nav
    class="_subheader_8b03fa"
  >
    <div
      class="_container_8b03fa"
    >
      <ul>
        <li>
          <a
            href="/category/all"
          >
            Shop All
          </a>
        </li>
        <li>
          <a
            href="/category/games"
          >
            Games
          </a>
        </li>
        <li>
          <a
            href="/category/plushies"
          >
            Plushies
          </a>
        </li>
        <li>
          <a
            href="/category/keychains"
          >
            Keychains
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;

exports[`Subheader > DOM > matches snapshot 1`] = `
<div>
  <nav
    class="_subheader_8b03fa"
  >
    <div
      class="_container_8b03fa"
    >
      <ul>
        <li>
          <a
            href="/category/all"
          >
            Shop All
          </a>
        </li>
        <li>
          <a
            href="/category/games"
          >
            Games
          </a>
        </li>
        <li>
          <a
            href="/category/plushies"
          >
            Plushies
          </a>
        </li>
        <li>
          <a
            href="/category/keychains"
          >
            Keychains
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
